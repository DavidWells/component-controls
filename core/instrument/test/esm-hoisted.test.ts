import { loadStoriesTests } from './loadTestFiles';

describe('esm-hoisted', () => {
  loadStoriesTests({ stories: { storeSourceFile: false } }, ['esm', 'hoisted']);
});
