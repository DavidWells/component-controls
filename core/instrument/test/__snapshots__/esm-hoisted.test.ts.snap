// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`esm-hoisted name-and-parameters.js 1`] = `
Object {
  "components": Object {
    "57e65d42bacf0af9b3f4dd3fe2377cde": Object {
      "name": "ControlsTable",
      "package": "c530e9236c215c5023fe34f23def3047",
    },
  },
  "doc": Object {
    "components": Object {},
    "componentsLookup": Object {
      "ControlsTable": "57e65d42bacf0af9b3f4dd3fe2377cde",
    },
    "date": 2020-07-17T14:45:45.643Z,
    "dateModified": 2020-07-17T14:45:45.643Z,
    "package": "c530e9236c215c5023fe34f23def3047",
    "title": "Story",
  },
  "packages": Object {
    "c530e9236c215c5023fe34f23def3047": Object {
      "dependencies": Object {
        "@babel/generator": "^7.9.4",
        "@babel/parser": "^7.9.4",
        "@babel/traverse": "^7.9.0",
        "@component-controls/core": "^1.17.0",
        "@hutson/parse-repository-url": "^5.0.0",
        "@mdx-js/loader": "^1.5.5",
        "@mdx-js/react": "^1.6.5",
        "camelcase": "^6.0.0",
        "deepmerge": "^4.2.2",
        "find-cache-dir": "^3.3.1",
        "gray-matter": "^4.0.2",
        "hosted-git-info": "^3.0.4",
        "js-string-escape": "^1.0.1",
        "prettier": "^1.19.1",
        "read-package-json": "^2.1.1",
        "remark-emoji": "^2.1.0",
        "remark-images": "^2.0.0",
        "resolve": "^1.15.1",
        "typescript": "^3.8.3",
      },
      "devDependencies": Object {
        "@babel/types": "^7.9.0",
        "@component-controls/ts-markdown-docs": "^1.10.0",
        "@rollup/plugin-node-resolve": "^7.1.1",
        "@types/find-cache-dir": "^3.2.0",
        "@types/hosted-git-info": "^3.0.0",
        "@types/jest": "^25.1.2",
        "cross-env": "^5.2.1",
        "eslint": "^6.5.1",
        "jest": "^24.9.0",
      },
      "fileHash": "c530e9236c215c5023fe34f23def3047",
      "name": "@component-controls/instrument",
      "peerDependencies": undefined,
      "repository": Object {
        "browse": "https://github.com/ccontrols/component-controls/tree/master/core/instrument/test/fixtures/esm/hoisted/name-and-parameters.js",
        "docs": "https://github.com/ccontrols/component-controls/tree/master#readme",
        "issues": "https://github.com/ccontrols/component-controls/issues",
      },
      "version": "1.17.0",
    },
  },
  "stories": Object {
    "myStory": Object {
      "arguments": Array [],
      "component": "ControlsTable",
      "id": "myStory",
      "loc": Object {
        "end": Object {
          "column": 31,
          "line": 2,
        },
        "start": Object {
          "column": 23,
          "line": 2,
        },
      },
      "name": "Custom story name",
      "smartControls": Object {
        "smart": false,
      },
      "source": "() => {}",
    },
  },
  "transformed": "export default { title: 'Story' };
export const myStory = () => {};
myStory.storyName = 'Custom story name';
myStory.component = ControlsTable;
myStory.smartControls = {
  smart: false,
};
",
}
`;

exports[`esm-hoisted name-parameters-and-controls.js 1`] = `
Object {
  "components": Object {},
  "doc": Object {
    "components": Object {},
    "componentsLookup": Object {},
    "date": 2020-07-17T14:45:45.643Z,
    "dateModified": 2020-07-17T14:45:45.644Z,
    "package": "9b19ea69af761f36bd07998cc518decf",
    "title": "Story",
  },
  "packages": Object {
    "9b19ea69af761f36bd07998cc518decf": Object {
      "dependencies": Object {
        "@babel/generator": "^7.9.4",
        "@babel/parser": "^7.9.4",
        "@babel/traverse": "^7.9.0",
        "@component-controls/core": "^1.17.0",
        "@hutson/parse-repository-url": "^5.0.0",
        "@mdx-js/loader": "^1.5.5",
        "@mdx-js/react": "^1.6.5",
        "camelcase": "^6.0.0",
        "deepmerge": "^4.2.2",
        "find-cache-dir": "^3.3.1",
        "gray-matter": "^4.0.2",
        "hosted-git-info": "^3.0.4",
        "js-string-escape": "^1.0.1",
        "prettier": "^1.19.1",
        "read-package-json": "^2.1.1",
        "remark-emoji": "^2.1.0",
        "remark-images": "^2.0.0",
        "resolve": "^1.15.1",
        "typescript": "^3.8.3",
      },
      "devDependencies": Object {
        "@babel/types": "^7.9.0",
        "@component-controls/ts-markdown-docs": "^1.10.0",
        "@rollup/plugin-node-resolve": "^7.1.1",
        "@types/find-cache-dir": "^3.2.0",
        "@types/hosted-git-info": "^3.0.0",
        "@types/jest": "^25.1.2",
        "cross-env": "^5.2.1",
        "eslint": "^6.5.1",
        "jest": "^24.9.0",
      },
      "fileHash": "9b19ea69af761f36bd07998cc518decf",
      "name": "@component-controls/instrument",
      "peerDependencies": undefined,
      "repository": Object {
        "browse": "https://github.com/ccontrols/component-controls/tree/master/core/instrument/test/fixtures/esm/hoisted/name-parameters-and-controls.js",
        "docs": "https://github.com/ccontrols/component-controls/tree/master#readme",
        "issues": "https://github.com/ccontrols/component-controls/issues",
      },
      "version": "1.17.0",
    },
  },
  "stories": Object {
    "myStory": Object {
      "arguments": Array [],
      "controls": Object {
        "name": Object {
          "label": "Name",
          "order": 9999,
          "type": "ControlTypes.TEXT",
          "value": "Mark",
        },
      },
      "id": "myStory",
      "loc": Object {
        "end": Object {
          "column": 31,
          "line": 2,
        },
        "start": Object {
          "column": 23,
          "line": 2,
        },
      },
      "name": "Custom story name",
      "smartControls": Object {
        "smart": false,
      },
      "source": "() => {}",
    },
  },
  "transformed": "export default { title: 'Story' };
export const myStory = () => {};
myStory.storyName = 'Custom story name';
myStory.controls = {
  name: {
    type: ControlTypes.TEXT,
    label: 'Name',
    value: 'Mark',
    order: 9999,
  },
};
myStory.smartControls = {
  smart: false,
};
",
}
`;

exports[`esm-hoisted story-property-name.js 1`] = `
Object {
  "components": Object {},
  "doc": Object {
    "components": Object {},
    "componentsLookup": Object {},
    "date": 2020-07-17T14:45:45.644Z,
    "dateModified": 2020-07-17T14:45:45.644Z,
    "package": "bb3ac5be32f72100f7edcb5a37624298",
    "title": "Story",
  },
  "packages": Object {
    "bb3ac5be32f72100f7edcb5a37624298": Object {
      "dependencies": Object {
        "@babel/generator": "^7.9.4",
        "@babel/parser": "^7.9.4",
        "@babel/traverse": "^7.9.0",
        "@component-controls/core": "^1.17.0",
        "@hutson/parse-repository-url": "^5.0.0",
        "@mdx-js/loader": "^1.5.5",
        "@mdx-js/react": "^1.6.5",
        "camelcase": "^6.0.0",
        "deepmerge": "^4.2.2",
        "find-cache-dir": "^3.3.1",
        "gray-matter": "^4.0.2",
        "hosted-git-info": "^3.0.4",
        "js-string-escape": "^1.0.1",
        "prettier": "^1.19.1",
        "read-package-json": "^2.1.1",
        "remark-emoji": "^2.1.0",
        "remark-images": "^2.0.0",
        "resolve": "^1.15.1",
        "typescript": "^3.8.3",
      },
      "devDependencies": Object {
        "@babel/types": "^7.9.0",
        "@component-controls/ts-markdown-docs": "^1.10.0",
        "@rollup/plugin-node-resolve": "^7.1.1",
        "@types/find-cache-dir": "^3.2.0",
        "@types/hosted-git-info": "^3.0.0",
        "@types/jest": "^25.1.2",
        "cross-env": "^5.2.1",
        "eslint": "^6.5.1",
        "jest": "^24.9.0",
      },
      "fileHash": "bb3ac5be32f72100f7edcb5a37624298",
      "name": "@component-controls/instrument",
      "peerDependencies": undefined,
      "repository": Object {
        "browse": "https://github.com/ccontrols/component-controls/tree/master/core/instrument/test/fixtures/esm/hoisted/story-property-name.js",
        "docs": "https://github.com/ccontrols/component-controls/tree/master#readme",
        "issues": "https://github.com/ccontrols/component-controls/issues",
      },
      "version": "1.17.0",
    },
  },
  "stories": Object {
    "myStory": Object {
      "arguments": Array [],
      "id": "myStory",
      "loc": Object {
        "end": Object {
          "column": 31,
          "line": 2,
        },
        "start": Object {
          "column": 23,
          "line": 2,
        },
      },
      "name": "myStory",
      "source": "() => {}",
    },
  },
  "transformed": "export default { title: 'Story' };
export const myStory = () => {};
story.storyName = 'Custom story name';
",
}
`;
